app/services/CheckoutService.scala

```diff
@@ -121,17 +122,22 @@ class CheckoutService(
     val idMinimalUser = authenticatedUserOpt.map(_.user)
 
     val telephoneNumber = subscriptionData.productData match {
-      case Left(paperData) => NormalisedTelephoneNumber.fromStringAndCountry(personalData.telephoneNumber, paperData.deliveryAddress.country orElse personalData.address.country)
+      case Left(paperData) => NormalisedTelephoneNumber.fromStringAndCountry(personalData.telephoneNumber, personalData.address.country orElse paperData.deliveryRecipient.address.country)
```

Comment:
Looking at the form, I think the phone number always belongs to the buyer, so I don't understand why we would ever use data about the delivery recipient?

Reply:
Because the personalData.address is null at this point if the user ticks the 'Use Delivery Address as billing' address. Hence the old logic was a bug giving higher precedence to the delivery country than the billing country when determining the phone number prefix.

Reply:
I'm confused as to where the billing country will be at this point..
but if we should give higher precedence to the billing country why would we ever check the delivery country? isn't billing country supposed to always be there ?